Fixes item embedding not working.

phil235 opened the pull request:
- Fixes #11117
- Fixes not getting the 'hit' message, damage and sound (and logging) when the item embeds itself in a human.
- Throwing a big item at an alien no longer pushes it.
- If a thrown item embeds itself in you, it doesn't also push you.
- Fixes an old argument in item/throw_impact() that I forgot to remove.
- Rename the skip argument of hitby() to skipcatch (more explicit).

phil235 committed:
Fixes item embedding not working.

Ergovisavi commented:
Does this also fix embedded objects not applying their throwforce before embedding?

phil235 commented:
@Ergovisavi it will after the next push.

phil235 committed:
Fixes not getting the hit message and sound (and logging) when the item embeds itself in a human.
Throwing a big item at an alien no longer pushes it.
If the item embeds itself in you, it doesn't push you.
Fixes an old argument in item/throw_impact() that I forgot to remove.
Rename the skip argument of hitby() to skipcatch (more explicit).
